src/uu/snice/src/snice.rs

```diff
@@ -0,0 +1,302 @@
+            Some(expr) => {
+                Priority::try_from(expr).map_err(|err| USimpleError::new(1, err.to_string()))?
+            }
+            None => Priority::default(),
+        };
+
+        let display = if matches.get_flag("table") {
+            Some(SignalDisplay::Table)
+        } else if matches.get_flag("list") {
+            Some(SignalDisplay::List)
+        } else {
+            None
+        };
+
+        Ok(Self {
+            display,
+            expressions: Self::targets(matches),
+            priority: expression,
+            verbose: matches.get_flag("verbose"),
+        })
+    }
+
+    fn targets(matches: &ArgMatches) -> Option<Vec<SelectedTarget>> {
+        let cmd = matches
+            .get_many::<String>("command")
+            .unwrap_or_default()
+            .map(Into::into)
+            .map(SelectedTarget::Command)
+            .collect::<Vec<_>>();
+
+        let pid = matches
+            .get_many::<u32>("pid")
+            .unwrap_or_default()
+            .map(Clone::clone)
+            .map(SelectedTarget::Pid)
+            .collect::<Vec<_>>();
+
+        let tty = matches
+            .get_many::<String>("tty")
+            .unwrap_or_default()
+            .flat_map(|it| Teletype::try_from(it.as_str()))
+            .map(SelectedTarget::Tty)
+            .collect::<Vec<_>>();
+
+        let user = matches
+            .get_many::<String>("user")
+            .unwrap_or_default()
+            .map(Into::into)
+            .map(SelectedTarget::User)
+            .collect::<Vec<_>>();
+
+        let collected = cmd
+            .into_iter()
+            .chain(pid)
+            .chain(tty)
+            .chain(user)
+            .collect::<Vec<_>>();
+
+        if collected.is_empty() {
+            None
+        } else {
+            Some(collected)
+        }
+    }
+}
+
+#[uucore::main]
+pub fn uumain(args: impl uucore::Args) -> UResult<()> {
+    let matches = uu_app().try_get_matches_from(args)?;
+
+    let settings = Settings::try_new(&matches)?;
+
+    // Case0: Print SIGNALS
+    #[cfg(target_family = "unix")]
+    {
+        if let Some(display) = settings.display {
+            let result = display.display(&ALL_SIGNALS);
+
+            println!("{result}");
+            return Ok(());
+        }
+    }
+
+    #[cfg(not(target_family = "unix"))]
+    {
+        if let Some(_display) = settings.display {
+            return Ok(());
+        }
+    }
+
+    // Case1: Perform priority
+    if let Some(targets) = settings.expressions {
+        let pids = collect_pids(&targets);
+        let results = perform_action(&pids, &settings.priority);
+
+        if results.iter().all(|it| it.is_none()) || results.is_empty() {
+            return Err(USimpleError::new(1, "no process selection criteria"));
+        }
+
+        let output = construct_verbose_result(&pids, &results).trim().to_owned();
+        if settings.verbose {
+            println!("{}", output);
```

Comment:
You only have to construct the verbose result if the "verbose" flag is set.
```suggestion
        if settings.verbose {
            let output = construct_verbose_result(&pids, &results).trim().to_owned();
            println!("{}", output);
```